app/src/main/kotlin/com/bitwarden/authenticator/ui/authenticator/feature/itemlisting/ItemListingViewModel.kt

```diff
@@ -261,32 +265,120 @@ class ItemListingViewModel @Inject constructor(
             when (val totpResult = action.totpResult) {
                 TotpCodeResult.CodeScanningError -> {
-                    sendAction(
-                        action = ItemListingAction.Internal.CreateItemResultReceive(
-                            result = CreateItemResult.Error,
-                        ),
-                    )
+                    handleCodeScanningErrorReceive()
                 }
 
-                is TotpCodeResult.Success -> {
-
-                    val item = totpResult.code.toAuthenticatorEntityOrNull()
-                        ?: run {
-                            sendAction(
-                                action = ItemListingAction.Internal.CreateItemResultReceive(
-                                    result = CreateItemResult.Error,
-                                ),
-                            )
-                            return@launch
-                        }
+                is TotpCodeResult.TotpCodeScan -> {
+                    handleTotpCodeScanReceive(totpResult)
+                }
 
-                    val result = authenticatorRepository.createItem(item)
-                    sendAction(ItemListingAction.Internal.CreateItemResultReceive(result))
+                is TotpCodeResult.GoogleExportScan -> {
+                    handleGoogleExportScan(totpResult)
                 }
             }
         }
     }
 
+    private suspend fun handleTotpCodeScanReceive(
+        totpResult: TotpCodeResult.TotpCodeScan,
+    ) {
+        val item = totpResult.code.toAuthenticatorEntityOrNull()
+            ?: run {
+                handleCodeScanningErrorReceive()
+                return
+            }
+
+        val result = authenticatorRepository.createItem(item)
+        sendAction(ItemListingAction.Internal.CreateItemResultReceive(result))
+    }
+
+    private suspend fun handleGoogleExportScan(
+        totpResult: TotpCodeResult.GoogleExportScan,
+    ) {
+        val base64EncodedMigrationData = encodingManager.uriDecode(
+            value = totpResult.data,
+        )
+
+        val decodedMigrationData = encodingManager.base64Decode(
+            value = base64EncodedMigrationData,
+        )
+
+        val payload = GoogleAuthenticatorProtos.MigrationPayload
+            .parseFrom(decodedMigrationData)
+
+        val entries = payload
+            .otpParametersList
+            .mapNotNull { otpParam ->
+                val secret = encodingManager.base32Encode(
+                    byteArray = otpParam.secret.toByteArray()
+                )
+
+                // Google Authenticator only supports TOTP and HOTP codes. We do not support HOTP
+                // codes so we skip over codes that are not TOTP.
+                val type = when (otpParam.type) {
+                    GoogleAuthenticatorProtos.MigrationPayload.OtpType.OTP_TOTP -> {
+                        AuthenticatorItemType.TOTP
+                    }
+
+                    else -> return@mapNotNull null
```

Comment:
❓ Not sure if it's a plausible scenario but, if so, shall we provide a specific error to the user when there are no codes in the payload with TOTP type? So it's easier for them to narrow down what went wrong.

Reply:
It's totally plausible if a user only enters HOTP codes into Google Authenticator. I like your idea, and easy enough to achieve. I have another PR on deck that adds a Google option to the Import screen. I'll add your suggested notification there.